File: Problem 1/firstProblem.py
# https://projecteuler.net/problem=1
def main(num_1, num_2, start_val, end_val):
	nums_to_check = range(start_val, end_val)
	first_num_list = []
	second_num_list = []
	third_num_list = []
	for num in nums_to_check:
		if num%num_1 == 0:
			first_num_list.append(num)
		if num%num_2 == 0:
			second_num_list.append(num)
		if num%(num_1*num_2) == 0:
			third_num_list.append(num)
	print(first_num_list)
	print(second_num_list)
	print(third_num_list)
	return sum(first_num_list) + sum(second_num_list) - sum(third_num_list)



if __name__ == "__main__":
    print(main(3, 5, 1, 1000))

File: Problem 1/firstProblem.jl
# https://projecteuler.net/problem=1

# If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

# Find the sum of all the multiples of 3 or 5 below 1000.
function sumDivs(num1::Int64, num2::Int64)
    howMany = Int64(floor((num2-1)/num1))
    top = howMany * num1
    return Int64((num1 + top) * (howMany/2))
end

function main(num1::Int64, num2::Int64, num::Int64)
    return sumDivs(num1, num)+sumDivs(num2, num) - sumDivs(num1*num2, num)
end
